sdk/framework/openapi.go

```diff
@@ -401,19 +380,40 @@ func documentPath(p *Path, backend *Backend, requestResponsePrefix string, doc *
 						},
 					}
 				}
-			}
 
-			// LIST is represented as GET with a `list` query parameter. Code later on in this function will assign
-			// list operations to a path with an extra trailing slash, ensuring they do not collide with read
-			// operations.
-			if opType == logical.ListOperation {
+			// For the operation types which map to HTTP methods without a request body, populate query parameters
+			case logical.ListOperation:
+				// LIST is represented as GET with a `list` query parameter. Code later on in this function will assign
+				// list operations to a path with an extra trailing slash, ensuring they do not collide with read
+				// operations.
 				op.Parameters = append(op.Parameters, OASParameter{
 					Name:        "list",
 					Description: "Must be set to `true`",
 					Required:    true,
 					In:          "query",
 					Schema:      &OASSchema{Type: "string", Enum: []interface{}{"true"}},
 				})
+				fallthrough
+			case logical.DeleteOperation:
+				fallthrough
+			case logical.ReadOperation:
+				for name, field := range queryFields {
+					t := convertType(field.Type)
+					p := OASParameter{
+						Name:        name,
+						Description: cleanString(field.Description),
+						In:          "query",
+						Schema: &OASSchema{
+							Type:         t.baseType,
+							Pattern:      t.pattern,
+							Enum:         field.AllowedValues,
+							Default:      field.Default,
+							DisplayAttrs: withoutOperationHints(field.DisplayAttrs),
+						},
+						Deprecated: field.Deprecated,
+					}
+					op.Parameters = append(op.Parameters, p)
+				}
```

Comment:
Note for reviewer: This added code is an approximate copy of the version handling path parameters at the top of this function.